Sources/MapboxCoreNavigation/MapboxRoutingProvider.swift

```diff
@@ -0,0 +1,338 @@
+@_implementationOnly import MapboxCommon_Private
+import MapboxDirections
+import MapboxNavigationNative
+@_implementationOnly import MapboxNavigationNative_Private
+
+/**
+ Provides alternative access to routing API.
+ 
+ Use this class instead `Directions` requests wrapper to request new routes or refresh an existing one. Depending on `RouterSource`, `MapboxRoutingProvider` will use online and/or onboard routing engines. This may be used when designing purely online or offline apps, or when you need to provide best possible service regardless of internet collection.
+ */
+public class MapboxRoutingProvider: RoutingProvider {
+    /**
+     Unique identifier for a giver request.
+     
+     Valid only for the same instance of `MapboxRoutingProvider` that issued it.
+     */
+    public typealias RequestId = UInt64
+    
+    /**
+     A request handler for the ongoing router action.
+     
+     You can use this instance to cancel ongoing task if needed. Retaining this handler will keep related `MapboxRoutingProvider` from deallocating.
+     */
+    public struct Request: NavigationProviderRequest {
+        /**
+         Related request identifier.
+         */
+        public let requestIdentifier: RequestId
+        
+        // Intended retain cycle to prevent deallocating. `Request` will be deleted once request completes.
+        let router: MapboxRoutingProvider
+        
+        /**
+         Cancels the request if it is still active.
+         */
+        public func cancel() {
+            router.router.cancelRequest(forToken: requestIdentifier)
+        }
+    }
+    
+    /**
+     Defines source of routing engine to be used for requests.
+     */
+    public enum Source {
```

Comment:
I don't see any new source types in the future. Given the domain of that feature, I believe it will be a significant change if we invents something new here